### PR TITLE
[rampup] Add o1js-stub to Mina repo in place of SnarkyJS 

### PR DESCRIPTION
**Pull request by MartinMinkov**
# Summary

o1js: https://github.com/o1-labs/o1js/pull/1358
bindings: https://github.com/o1-labs/o1js-bindings/pull/232

Addresses https://github.com/MinaProtocol/mina/issues/14466

This PR aims to add support for building mina as a submodule of `o1js`. To do so, the following was done:
1. Removed the `snarkyjs` submodule
2. Added a `o1js_stub` module, that has the same dependencies as o1js. The goal with this module is to have a way for Mina CI to check that o1js can still be successfully compiled after changes are made. This gives a safety net to not totally break things if changes to the dependencies are made.
3. Adds the `kimchi` library from o1js-bindings into `src/lib/crypto`. This is to make sure that we can compile `o1js_stub` since we need the WASM dependencies generated from the library, as well as to support a longer term plan to create a o1 crypto repo that recombines the wasm bindings with o1caml-rust stubs.
4. Removes snarkyjs from nix and updates the Rust part of nix to reference the new kimchi library location
5. Removed codeowners entry for `snarkyjs`

### CODE REVIEW

**Comment by MartinMinkov** [comment]
!ci-build-me

**Comment by MartinMinkov** [comment]
!ci-build-me

**Comment by MartinMinkov** [comment]
!ci-build-me

**Comment by MartinMinkov** [comment]
!ci-build-me

**Comment by MartinMinkov** [comment]
!ci-build-me

File: dockerfiles/stages/1-build-deps [hunk 1]
```diff
@@ -119,8 +120,8 @@ RUN git clone https://github.com/facebook/rocksdb \
   --depth 1 --shallow-submodules \
   -b "${ROCKSDB_VERSION}" /rocksdb \
   && test $(g++ -dumpversion | cut -c 1 -) -lt 9 \
-    && CXXFLAGS='' make -C /rocksdb static_lib PORTABLE=1 -j$(nproc) 2>/dev/null \
-    || make -C /rocksdb static_lib PORTABLE=1 -j$(nproc) 2>/dev/null \
+  && CXXFLAGS='' make -C /rocksdb static_lib PORTABLE=1 -j$(nproc) 2>/dev/null \
```
Comment (mrmr1993) [inline dockerfiles/stages/1-build-deps:123]: Revert

File: dockerfiles/stages/3-builder [hunk 2]
```diff
@@ -36,23 +36,25 @@ RUN mkdir ${HOME}/app
...
-    src/app/archive_blocks/archive_blocks.exe \
-    src/app/extract_blocks/extract_blocks.exe \
-    src/app/missing_blocks_auditor/missing_blocks_auditor.exe \
-    src/app/replayer/replayer.exe \
-    src/app/rosetta/rosetta_testnet_signatures.exe \
-    src/app/rosetta/rosetta_mainnet_signatures.exe \
-    src/app/generate_keypair/generate_keypair.exe \
-    src/app/validate_keypair/validate_keypair.exe \
-    src/app/rosetta/ocaml-signer/signer.exe \
+  src/app/cli/src/mina_testnet_signatures.exe \
```
Comment (mrmr1993) [inline dockerfiles/stages/3-builder:45]: Revert

**Review by mrmr1993 — APPROVED** [review]
LGTM, but please revert unrelated script indentation changes.

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!approved-for-mainnet

**Comment by MartinMinkov** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!approved-for-mainnet

**Review by mrmr1993 — APPROVED** [review]

**Review by bkase — APPROVED** [review]

**Pull request merged**